### PR TITLE
Hoverable dropdown navigation

### PR DESCRIPTION
**Pull request by geek-yang**
- Add navigation buttons in the banner on top of each page
- Create dropdown menu for subpages and add navigation
- Hoverable dropdown menu design

### CODE REVIEW

File: components/Navigation.vue [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+<script>
+export default {
+  data () {
+    return {
+      hover: false,
+      pages: [
+        {
+          parent: { title: 'CPM', url: '/cpm' },
+          children: [{ title: 'Explore', url: '/cpm/explore' },
```
Comment (Peter9192) [inline components/Navigation.vue:46]: The few trees I've recently encountered seem to follow a structure along these lines:

```
pages: [
  { 
    title: 'CPM', 
    url: '/cpm', 
    children: [
      {
        title: 'Explore',
        url: 'cpm/explore',
        children: {}
      },
},
```

Note that having the `parent` attribute is not necessary. By leaving it out, the children become identical to the parent, which makes it possible to make endlessly nested trees that can be traversed tree recursively, e.g. like so:

```js
function traverse (tree) {
    if (tree.children != {}) {
      tree.forEach(node => traverse(node.children))
    } else {
      // This is a leaf node.
    }
}
```

File: components/Navigation.vue [hunk 2]
```diff
@@ -0,0 +1,67 @@
+<template>
+  <div class="flex ml-auto gap-10 m-2 mr-20 space-x-1 text-l">
```
Comment (Peter9192) [inline components/Navigation.vue:2]:
```suggestion
  <div class="flex ml-auto gap-2 m-2 mr-20 space-x-1 text-l">
```
Could be personal preference

File: components/Navigation.vue [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+      <NuxtLink :to="page.parent.url">
+        <button
+          class="w-32 bg-blue-300 hover:bg-blue-500 text-white py-2 px-4 rounded"
+        >
+          {{ page.parent.title }}
+        </button>
+      </NuxtLink>
+      <!-- dropdown pages -->
+      <div
+        v-if="page.hover"
```
Comment (Peter9192) [inline components/Navigation.vue:20]: Interesting read: https://learnvue.co/2021/05/v-if-vs-v-show-vue-conditional-rendering/#when-would-you-want-to-use-each-one

File: components/Navigation.vue [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  data () {
+    return {
+      hover: false,
```
Comment (Peter9192) [inline components/Navigation.vue:42]: Is `hover` (without `page.`) actually used?

File: components/Navigation.vue [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+            { title: 'Download', url: '/cpm/download' },
+            { title: 'More info', url: '/cpm/references' }],
+          hover: false
+        },
+        {
+          parent: { title: 'Decadal', url: '/decadal' },
+          children: [{ title: 'Forecasts', url: '/decadal/forecasts' },
+            { title: 'Hindcasts', url: '/decadal/hindcasts' },
+            { title: 'Experiments', url: '/decadal/experiments' }],
+          hover: false
```
Comment (Peter9192) [inline components/Navigation.vue:57]: Separate functionality from data: `hover` is a bit out of place here, and you need to type it every time and always fill in false. What about:

```
export default {
  data () {
    pages = [ ... ]
    pages.forEach(page => page.hover = false)
    return pages
  }
}
```

Still not ideal I'd say, but at least now I can easily modify the data without having to worry about the way it is used internally.

**Review by Peter9192 — APPROVED** [review]
Great work Yang! It works and I'm happy to merge it as is, but I have a few comments that I think you can still benefit from. Cheers!

File: components/Navigation.vue (hunk 1 above)
Comment (geek-yang) [reply components/Navigation.vue]: Looks really awesome! Just removed the redundant `parent` attribute.

File: components/Navigation.vue (hunk 2 above)
Comment (geek-yang) [reply components/Navigation.vue:2]: No problem! Corrected!

File: components/Navigation.vue (hunk 4 above)
Comment (geek-yang) [reply components/Navigation.vue:42]: No. Sorry that this was the variable used before. It is removed now.

File: components/Navigation.vue (hunk 5 above)
Comment (geek-yang) [reply components/Navigation.vue]: Nice suggestion. But it seems this syntax is not allowed. I added `this.pages = this.pages.map(page => ({ ...page, hover: false }))` in `mounted () {}` and it works well now.

**Pull request merged**